roles/openshift_version/tasks/main.yml

```diff
@@ -9,24 +9,43 @@
 # be used by default. Users must indicate what they want.
 - fail:
     msg: "Must specify openshift_release or openshift_image_tag in inventory to install origin. (suggestion: add openshift_release=\"1.2\" to inventory)"
-  when: is_containerized | bool and openshift.common.deployment_type == 'origin' and openshift_release is not defined and openshift_image_tag is not defined
+  when:
+  - is_containerized | bool
+  - openshift.common.deployment_type == 'origin'
+  - openshift_release is not defined
+  - openshift_image_tag is not defined
 
 # Normalize some values that we need in a certain format that might be confusing:
 - set_fact:
     openshift_release: "{{ openshift_release[1:] }}"
-  when: openshift_release is defined and openshift_release[0] == 'v'
+  when:
+  - openshift_release is defined
+  - openshift_release[0] == 'v'
 
 - set_fact:
     openshift_release: "{{ openshift_release | string }}"
   when: openshift_release is defined
 
+# Verify that the image tag is in a valid format
+- assert:
+    that:
+      - "{{ openshift_image_tag|match('(^[v]?\\d+\\.\\d+[\\.\\d+]*-?\\w*$)') }}"
+    msg: "openshift_image_tag must be in the format [v]#.#[.#...]. Examples: 1.2, v3.4.1, 3.5.1.3"
+  when: openshift_image_tag is defined
+
+# Add a v to the start of the openshift_image_tag if one was not provided
 - set_fact:
     openshift_image_tag: "{{ 'v' + openshift_image_tag }}"
-  when: openshift_image_tag is defined and openshift_image_tag[0] != 'v' and openshift_image_tag != 'latest'
+  when:
+  - openshift_image_tag is defined
+  - openshift_image_tag[0] != 'v'
+  - openshift_image_tag != 'latest'
 
 - set_fact:
     openshift_pkg_version: "{{ '-' + openshift_pkg_version }}"
-  when: openshift_pkg_version is defined and openshift_pkg_version[0] != '-'
```

Comment:
It's a little too late for this ... but it seems like exposing a very small surface for these vars would be better than trying to accept the world and deal with it.